tuple.go

```diff
@@ -0,0 +1,424 @@
+					if field.Kind() == reflect.Struct {
+						EncodeTuple(field.Interface(), buffer)
+					} else {
+						panic("unreachable case (Struct) in EncodeTuple")
+					}
+				}
+			case reflect.Float32:
+				panic("encoding of Float32 field is not supported")
+			case reflect.Float64:
+				panic("encoding of Float64 field is not supported")
+			case reflect.Complex64:
+				panic("encoding of Complex64 field is not supported")
+			case reflect.Complex128:
+				panic("encoding of Complex128 field is not supported")
+			case reflect.Uintptr:
+				panic("encoding of Uintptr field is not implemented")
+			case reflect.UnsafePointer:
+				panic("encoding of UnsafePointer field is not implemented")
+			case reflect.Pointer:
+				panic("encoding of Pointer field is not implemented")
+			case reflect.Chan:
+				panic("encoding of Chan field is not implemented")
+			case reflect.Func:
+				panic("encoding of Func field is not implemented")
+			case reflect.Interface:
+				/*
+					Here it does nothing, but that allows the usage of the embedded Encodable
+					in custom-defined structs which allows using them in places where Encodable
+					is expected like in the case of Option[T], Result[T].
+				*/
+			default:
+				panic("unreachable case in EncodeTuple")
+			}
+
+		}
+	}
+}
+
+func SequenceFieldEncode(field reflect.Value, buffer *bytes.Buffer) {
+	// Tinygo does not support: reflect.SliceOf(field.Type())
+	switch field.Type().Elem() {
+	case reflect.TypeOf(*new(Bool)):
+		ConvertToSequence[Bool](field).Encode(buffer)
+	case reflect.TypeOf(*new(U8)):
+		ConvertToSequence[U8](field).Encode(buffer)
+	case reflect.TypeOf(*new(I8)):
+		ConvertToSequence[I8](field).Encode(buffer)
+	case reflect.TypeOf(*new(U16)):
+		ConvertToSequence[U16](field).Encode(buffer)
+	case reflect.TypeOf(*new(I16)):
+		ConvertToSequence[I16](field).Encode(buffer)
+	case reflect.TypeOf(*new(U32)):
+		ConvertToSequence[U32](field).Encode(buffer)
+	case reflect.TypeOf(*new(I32)):
+		ConvertToSequence[I32](field).Encode(buffer)
+	case reflect.TypeOf(*new(U64)):
+		ConvertToSequence[U64](field).Encode(buffer)
+	case reflect.TypeOf(*new(I64)):
+		ConvertToSequence[I64](field).Encode(buffer)
+	case reflect.TypeOf(*new(U128)):
+		ConvertToSequence[U128](field).Encode(buffer)
+	case reflect.TypeOf(*new(I128)):
+		ConvertToSequence[I128](field).Encode(buffer)
+	case reflect.TypeOf(*new(Compact)):
+		ConvertToSequence[Compact](field).Encode(buffer)
+	case reflect.TypeOf(*new(CompactU128)):
+		ConvertToSequence[CompactU128](field).Encode(buffer)
+	case reflect.TypeOf(*new(Str)):
+		ConvertToSequence[Str](field).Encode(buffer)
+	case reflect.TypeOf(*new(VaryingData)):
+		ConvertToSequence[VaryingData](field).Encode(buffer)
+
+	// TODO: Sequence[Sequence[T]]
+	// TODO: Sequence[FixedSequence[T]]
+	// TODO: Sequence[Dictionary[T1, T2]]
+
+	default:
+
+		switch field.Type() {
+		case reflect.TypeOf(*new(VaryingData)):
+			ConvertTo[VaryingData](field).Encode(buffer)
```

Comment:
Is this needed?

Reply:
yes, here  `func Test_EncodeTupleVaryingData(t *testing.T) {`